syntax = "proto2";

package lego.network.protobuf;

message GetNetworkNodesRequest {
	optional uint32 net_id = 1;
	optional uint32 count = 2;
	optional uint32 country = 3;
}

message NodeInfo {
    optional bytes public_ip = 1;
    optional int32 public_port = 2;
    optional bytes local_ip = 3;
    optional int32 local_port = 4;
    optional bytes id = 5;
    optional int32 nat_type = 6;
    optional bytes dht_key = 7;
    optional bytes pubkey = 8;
}

message GetNetworkNodesResponse {
    repeated NodeInfo nodes = 1;
}

message NetworkMessage {
	optional GetNetworkNodesRequest get_net_nodes_req = 1;
	optional GetNetworkNodesResponse get_net_nodes_res = 2;
}